pkg/networkdevice/profile/profiledefinition/metrics.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package profiledefinition
+
+import (
+	"regexp"
+)
+
+type ProfileMetricType string
+
+const (
+	ProfileMetricTypeGauge                 ProfileMetricType = "gauge"
+	ProfileMetricTypeMonotonicCount        ProfileMetricType = "monotonic_count"
+	ProfileMetricTypeMonotonicCountAndRate ProfileMetricType = "monotonic_count_and_rate"
+	ProfileMetricTypeRate                  ProfileMetricType = "rate"
+	ProfileMetricTypeFlagStream            ProfileMetricType = "flag_stream"
+
+	// ProfileMetricTypeCounter is DEPRECATED
+	// `counter` is deprecated in favour of `rate`
+	ProfileMetricTypeCounter ProfileMetricType = "counter"
+
+	// ProfileMetricTypePercent is DEPRECATED
+	// `percent` is deprecated in favour of `scale_factor`
+	ProfileMetricTypePercent ProfileMetricType = "percent"
+)
+
+// SymbolConfig holds info for a single symbol/oid
+type SymbolConfig struct {
+	OID  string `yaml:"OID" json:"OID"`
+	Name string `yaml:"name" json:"name"`
+
+	ExtractValue         string         `yaml:"extract_value" json:"extract_value"`
+	ExtractValueCompiled *regexp.Regexp // TODO: ADD `-` yaml/json annotation?
+
+	MatchPattern         string         `yaml:"match_pattern" json:"match_pattern"`
+	MatchValue           string         `yaml:"match_value" json:"match_value"`
+	MatchPatternCompiled *regexp.Regexp // TODO: ADD `-` yaml/json annotation?
```

Comment:
I'm curious about this pattern - as I understand it, the yaml and json modules both understand regexes, so you could just have `MatchPattern` be a regex and `yaml.Marshal` would convert it to a string while `yaml.Unmarshal` would parse it back to a regex (returning an error if it was malformed). Why do we keep these as two separate variables?

Reply:
Indeed, `MatchPatternCompiled *regexp.Regexp` and other fields without annotation are actually not meant to be serialised/deserialised.

They are used to internally store the compiled regexes.

I think we should add ``yaml:"-" json:"-"`` to ignore them (I can handle that as separate PR since not necessarily in the scope of this PR)

Reply:
I'm wondering why we have those fields at all - if we just changed the type of `MatchPattern` to be `*regexp.Regexp`, then the yaml and json parsers would compile the regex for us.

See e.g. https://play.golang.com/p/z2nl8woTnvf

Then we wouldn't need a separate step to go through and compile all the regexes.